rudder_airflow_provider/operators/rudderstack.py

```diff
@@ -24,9 +21,39 @@ def execute(self, context):
         '''
             Executes rudderstack operator
         '''
-        rs_hook = RudderstackHook(
-            connection_id=self.connection_id, source_id=self.source_id)
-        run_id = rs_hook.trigger_sync()
+        rs_hook = RudderstackHook(connection_id=self.connection_id)
+        run_id = rs_hook.trigger_sync(self.source_id)
         if self.wait_for_completion and run_id is not None:
             logging.info('waiting for sync to complete for sourceId: %s, runId: %s', self.source_id, run_id)
-            rs_hook.poll_for_status(run_id)
+            rs_hook.poll_for_status(self.source_id, run_id)
+
+
+class RudderstackRETLOperator(baseoperator.BaseOperator):
+    '''
+        Rudderstack operator for RETL connnections
+        :param retl_connection_id: unique id of the retl connection
+        :param sync_type: type of sync to trigger. Possible values are 'full' or 'incremental'
+        :param connection_id: airflow connection id for rudderstack API
+        :param wait_for_completion: wait for sync to complete. Default is False
+    '''
+    def __init__(self, 
+                 retl_connection_id: str, 
+                 sync_type: str, 
+                 connection_id: str = RUDDERTACK_DEFAULT_CONNECTION_ID,
+                 wait_for_completion: bool = False,
+                 **kwargs):
+        '''
+            Initialize rudderstack operator
+        '''
+        super().__init__(**kwargs)
+        self.wait_for_completion = wait_for_completion
+        self.connection_id = connection_id
+        self.retl_connection_id = retl_connection_id
+        self.sync_type = sync_type
```

Comment:
In the previous version, we don't have this. we expect full syncs to be triggered manually using the webapp. we can take a call on this whether to add  this in operator or not. 

If we decide to keep this, it should be templatable, so that this can be configured using some variable in airflow
https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html

Reply:
I don't see a case where this will be hardcoded to value like `full` | `incremental`

Reply:
Hardcoding to full atleast doesn't make sense. It can be decided manually or through code for particular run.

Reply:
made it templatable. can be changed whenever a full sync is required